girder/models/folder.py

```diff
@@ -514,29 +514,42 @@ def parentsToRoot(self, folder, curPath=None, user=None, force=False,
             return self.parentsToRoot(curParentObject, curPath, user=user,
                                       force=force)
 
-    def subtreeCount(self, folder):
+    def subtreeCount(self, folder, includeItems=True, user=None, level=None):
         """
         Return the size of the subtree rooted at the given folder. Includes
-        the root folder in the count. Counts folders and items. This returns the
-        absolute size of the subtree, it does not filter by permissions.
+        the root folder in the count.
 
         :param folder: The root of the subtree.
         :type folder: dict
+        :param includeItems: Whether to include items in the subtree count, or
+            just folders.
+        :type includeItems: bool
+        :param user: If filtering by permission, the user to filter against.
+        :param level: If filtering by permission, the required permission level.
+        :type level: AccessLevel
         """
         count = 1
 
-        items = self.model('item').find({
-            'folderId': folder['_id']
-        }, fields=())
-        count += items.count()
-        # subsequent operations take a long time, so free the cursor's resources
-        items.close()
+        if includeItems:
+            items = self.model('item').find({
```

Comment:
If you wanted to be premature, you could call count instead of find on the items, since we don't filter them by permission.

see http://api.mongodb.org/python/current/api/pymongo/collection.html#pymongo.collection.Collection.count